driver/pace/driver/run.py

```diff
@@ -116,85 +130,172 @@ class SerialboxConfig(InitializationConfig):
+        stencil_factory = StencilFactory(
+            config=stencil_config, grid_indexing=grid.grid_indexing
+        )
+        driver_grid_info_object = TranslateUpdateDWindsPhys(
+            grid, self.namelist, stencil_factory
+        )
+        extra_grid_data = driver_grid_info_object.collect_input_data(
+            ser, ser.get_savepoint("FVUpdatePhys-In")[0]
+        )
+        driver_grid_data = pace.util.grid.DriverGridData(
+            vlon1=extra_grid_data["vlon1"],
+            vlon2=extra_grid_data["vlon2"],
+            vlon3=extra_grid_data["vlon3"],
+            vlat1=extra_grid_data["vlat1"],
+            vlat2=extra_grid_data["vlat2"],
+            vlat3=extra_grid_data["vlat3"],
+            edge_vect_w=extra_grid_data["edge_vect_w"],
+            edge_vect_e=extra_grid_data["edge_vect_e"],
+            edge_vect_s=extra_grid_data["edge_vect_s"],
+            edge_vect_n=extra_grid_data["edge_vect_n"],
+            es1_1=extra_grid_data["es1_1"],
+            es1_2=extra_grid_data["es2_1"],
+            es1_3=extra_grid_data["es3_1"],
+            ew2_1=extra_grid_data["ew1_2"],
+            ew2_2=extra_grid_data["ew2_2"],
+            ew2_3=extra_grid_data["ew3_2"],
+        )
+        return grid, damping_coefficients, driver_grid_data
+
+    def _serializer(self, communicator: pace.util.CubedSphereCommunicator):
+        import serialbox
+
+        serializer = serialbox.Serializer(
+            serialbox.OpenModeKind.Read,
+            self.path,
+            "Generator_rank" + str(communicator.rank),
+        )
+        return serializer
+
+    def _get_grid_data_damping_coeff_and_driver_grid(
+        self,
+        quantity_factory: pace.util.QuantityFactory,
+        communicator: pace.util.CubedSphereCommunicator,
+    ):
+        if self.serialized_grid:
+            (
+                grid,
+                damping_coeff,
+                driver_grid_data,
+            ) = self._get_serialized_grid_damping_coeff_and_driver_grid(communicator)
+            grid_data = grid.grid_data
+        else:
+            grid = fv3core._config.make_grid_with_data_from_namelist(
+                self.namelist, communicator, self.backend
+            )
+            metric_terms = pace.util.grid.MetricTerms(
+                quantity_factory=quantity_factory, communicator=communicator
+            )
+            grid_data = pace.util.grid.GridData.new_from_metric_terms(metric_terms)
+            damping_coeff = DampingCoefficients.new_from_metric_terms(metric_terms)
+            driver_grid_data = pace.util.grid.DriverGridData.new_from_metric_terms(
+                metric_terms
+            )
+        return grid, grid_data, damping_coeff, driver_grid_data
+
     def get_driver_state(
         self,
         quantity_factory: pace.util.QuantityFactory,
         communicator: pace.util.CubedSphereCommunicator,
     ) -> DriverState:
-        raise NotImplementedError()
+        self.backend = quantity_factory.empty(
```

Comment:
Can you pass this as an explicit argument to the private methods that use it, instead of passing it through an object attribute? We don't need this data to persist on `self` between calls, and passing it as an argument makes this clearer.

Reply:
Done!